patches/chrome-browser-ui-toolbar-recent_tabs_sub_menu_model_unittest.cc.patch

```diff
@@ -1,32 +1,46 @@
 diff --git a/chrome/browser/ui/toolbar/recent_tabs_sub_menu_model_unittest.cc b/chrome/browser/ui/toolbar/recent_tabs_sub_menu_model_unittest.cc
-index b12ca875746ef64c05cbc9a1de0769a8f3c65f47..71da65021b83e37ef478963e3f9c1b654825232f 100644
+index f42476a2cc24c7d1744a5fa19b3cd2a7c487dc4e..46c8691d05175fe5678a57592a13b733318fd99c 100644
 --- a/chrome/browser/ui/toolbar/recent_tabs_sub_menu_model_unittest.cc
 +++ b/chrome/browser/ui/toolbar/recent_tabs_sub_menu_model_unittest.cc
-@@ -172,6 +172,7 @@ class RecentTabsSubMenuModelTest : public BrowserWithTestWindowTest {
-                              ->GetUnderlyingOpenTabsUIDelegateForTest());
+@@ -175,6 +175,7 @@ class RecentTabsSubMenuModelTest : public BrowserWithTestWindowTest,
+     helper->VerifyExport(session_sync_service_->GetOpenTabsUIDelegate());
    }
  
 +  BRAVE_RECENT_TABS_SUB_MENU_MODEL_TEST
   private:
    raw_ptr<sync_sessions::SessionSyncService, DanglingUntriaged>
        session_sync_service_;
-@@ -580,8 +581,8 @@ TEST_F(RecentTabsSubMenuModelTest, MaxSessionsAndRecency) {
-   EXPECT_THAT(base::span<const std::u16string>(
-                   recent_tabs_builder.GetTabTitlesSortedByRecency())
-                   .subspan(0, 3),
--              ElementsAre(model.GetLabelAt(5), model.GetLabelAt(8),
--                          model.GetLabelAt(11)));
-+              ElementsAre(model.GetLabelAt(6), model.GetLabelAt(9),
-+                          model.GetLabelAt(12)));
+@@ -643,15 +644,15 @@ TEST_P(RecentTabsSubMenuModelTest, MaxSessionsAndRecency) {
+     EXPECT_THAT(base::span<const std::u16string>(
+                     recent_tabs_builder.GetTabTitlesSortedByRecency())
+                     .subspan(0, 3),
+-                ElementsAre(model.GetLabelAt(6), model.GetLabelAt(9),
```

Comment:
why are we patching unit tests? @AlexeyBarabash @simonhong we should not be doing this. If we have different behavior we should disable the upstream tests and add our own in brave

Reply:
this should probably be a follow-up for someone outside of chromium updates

Reply:
@bridiver 

https://github.com/brave/brave-browser/issues/34658

My idea was to to minimize duplication of the code by re-using the Chromium's test. As it causes troubles regularly, I'll re-do.